src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -1066,4 +1076,24 @@ public String getJcrOutputPath() {
     }
   }
 
+  protected int checkAndGetUserInputQueryLimit( Object userInputQueryLimit, int reportQueryLimit ) {
+    int systemQueryLimit = -1;
```

Comment:
If a report-query-limit is defined, you MUST use this value. You *may* limit it by the system limit (pending today's meeting results). If there is a reportQueryLimit, the userQueryLimit must NOT be used.

Please do me a favour and add a unit-test for this method where you test out the various combinations. Remember: New code needs testing anyway, and with proper tests you will detect a few logic flaws here.